test/query_test.rb

```diff
@@ -343,137 +329,109 @@ def test_time
-    assert_query Time.at(secs).utc, Query.epoch(secs, 'second')
+    assert_equal Time.at(secs).utc, client.query { epoch(secs, 'second') }
     nanos = RandomHelper.random_number
-    assert_query Time.at(Rational(nanos, 1_000_000_000)).utc, Query.epoch(nanos, 'nanosecond')
+    assert_equal Time.at(Rational(nanos, 1_000_000_000)).utc, client.query { epoch(nanos, 'nanosecond') }
   end
 
   def test_date
-    assert_query Date.new(1970, 1, 1), Query.date('1970-01-01')
+    assert_equal Date.new(1970, 1, 1), client.query { date('1970-01-01') }
   end
 
   def test_equals
-    assert_query true, Query.equals(1, 1, 1)
-    assert_query false, Query.equals(1, 1, 2)
-    assert_query true, Query.equals(1)
-    assert_bad_query Query.equals
+    assert_equal true, client.query { equals(1, 1, 1) }
+    assert_equal false, client.query { equals(1, 1, 2) }
+    assert_equal true, client.query { equals(1) }
   end
 
   def test_contains
-    obj = Query.quote a: { b: 1 }
-    assert_query true, Query.contains([:a, :b], obj)
-    assert_query true, Query.contains(:a, obj)
-    assert_query false, Query.contains([:a, :c], obj)
+    obj = Fauna.query { quote({ a: { b: 1 } }) }
+    assert_equal true, client.query { contains([:a, :b], obj) }
+    assert_equal true, client.query { contains(:a, obj) }
+    assert_equal false, client.query { contains([:a, :c], obj) }
   end
 
   def test_select
-    obj = Query.quote a: { b: 1 }
-    assert_query({ b: 1 }, Query.select(:a, obj))
-    assert_query 1, Query.select([:a, :b], obj)
-    assert_query nil, Query.select(:c, obj, default: nil)
+    obj = Fauna.query { quote({ a: { b: 1 } }) }
+    assert_equal({ b: 1 }, client.query { select(:a, obj) })
+    assert_equal 1, client.query { select([:a, :b], obj) }
+    assert_equal nil, client.query { select(:c, obj, default: nil) }
     assert_raises(NotFound) do
-      client.query Query.select(:c, obj)
+      client.query { select(:c, obj) }
     end
   end
 
   def test_select_array
     arr = [1, 2, 3]
-    assert_query 3, Query.select(2, arr)
+    assert_equal 3, client.query { select(2, arr) }
     assert_raises(NotFound) do
-      client.query Query.select(3, arr)
+      client.query { select(3, arr) }
     end
   end
 
   def test_add
-    assert_query 10, Query.add(2, 3, 5)
-    assert_bad_query Query.add
+    assert_equal 10, client.query { add(2, 3, 5) }
   end
 
   def test_multiply
-    assert_query 30, Query.multiply(2, 3, 5)
-    assert_bad_query Query.multiply
+    assert_equal 30, client.query { multiply(2, 3, 5) }
   end
 
   def test_subtract
-    assert_query(-6, Query.subtract(2, 3, 5))
-    assert_query 2, Query.subtract(2)
-    assert_bad_query Query.subtract
+    assert_equal -6, client.query { subtract(2, 3, 5) }
+    assert_equal 2, client.query { subtract(2) }
   end
 
   def test_divide
-    assert_query 2.0 / 15, Query.divide(2.0, 3, 5)
-    assert_query 2, Query.divide(2)
-    assert_bad_query Query.divide(1, 0)
```

Comment:
yes. assuming the error handling mechanism is tested, behavioral tests of individual functions are out of scope for these.